### PR TITLE
commands: add commands for updating numeric scale for large tables

### PR DESCRIPTION
**Pull request by risicle**
Replaces #3952, an attempt to do this in migrations.

https://trello.com/c/cAARpLJ2/501-fix-numerics-which-were-converted-to-6dp-by-dms

These commands update the numerics in `Notification`, `NotificationHistory` and `FactBilling` to their `min_scale` following the DMS migration.

For `FactBilling` and `Notification`, performs updates in blocks (partitioned on a pseudorandom uuid) to avoid locking any whole tables (or even one specific time-period) at once.

For `NotificationHistory`, partition the blocks on `created_at` as that is more closely correlated to the physical layout of the table, and accesses to it (outside the periodic update tasks) aren't particularly clustered around any one point in time that we should avoid locking.

### CODE REVIEW

File: app/commands.py [hunk 1]
```diff
@@ -818,6 +829,115 @@ def local_dev_broadcast_permissions(user_id):
...
+                {
+                    "rate": _get_min_scale_cases(FactBilling.rate),
+                }
+            )
+
+
+@click.option("-h", "--block-hours", type=float, default=1.0)
+@notify_command(name="update-notification-history-numerics-min-scale")
+def update_notification_history_numerics_min_scale(block_hours):
+    block_period = timedelta(microseconds=block_hours * 60 * 60 * 1e3 * 1e3)
```
Comment (samuelhwilliams) [inline app/commands.py:900]:
```suggestion
    block_period = timedelta(hours=block_hours)
```

?

Comment (risicle) [reply app/commands.py:900]: `hours=` needs to be an integer, no?

Comment (risicle) [reply app/commands.py:900]: ...no they do not!

TIL

**Review by samuelhwilliams — APPROVED** [review]
These look good to me 👍  All work as expected locally and I think the strategies for grouping are sensible.

**Comment by samuelhwilliams** [comment]
You _may_ want to install the `pre-commit` hooks so you don't get CI test failures for formatting things.

**Pull request merged**